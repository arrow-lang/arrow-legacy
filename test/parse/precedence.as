// Identitiy, negation, bitwise not, dereference, and address-of [2]
+x;
-x;
!x;
*x;
&x;

// Multiplication, division, and remainder [3]
-x * y;
y / +x;
!y % x;

// Addition and subtraction [4]
y + x * z;
y - x % z;
(a - b) / y;

// Bitwise AND [5]
a & b;
a & b * x;
(a & b) * x;

// Bitwise XOR [6]
a ^ b;
b ^ c & d;

// Bitwise OR [7]
a ^ b | c;
b | c & b ^ a;

// Greater than; and, greater than or equal to [8]
x > y;
x >= y;

// Less than; and, less than or equal to [8]
y > x;
y >= x;

// Equal to; and, not equal to [10]
x == y;

// do_logical NOT [11]
not x == y;
not y > x;

// do_logical AND [12]
not x < y and false;

// do_logical OR [13]
not x < y or false;
y > x or x < y and y > 1;

// Assignment [15]
x = y or z;
