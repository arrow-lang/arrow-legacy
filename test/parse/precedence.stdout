{"tag":"Module","span":"test/parse/precedence.as:1.1-54.11","name":"precedence","block":{"tag":"Block","span":"test/parse/precedence.as:1.1-54.11","statements":[{"tag":"Identity","span":"test/parse/precedence.as:2.1-3","operand":{"tag":"Identifier","span":"test/parse/precedence.as:2.2-3","text":"x"}},{"tag":"Negate","span":"test/parse/precedence.as:3.1-3","operand":{"tag":"Identifier","span":"test/parse/precedence.as:3.2-3","text":"x"}},{"tag":"BitNot","span":"test/parse/precedence.as:4.1-3","operand":{"tag":"Identifier","span":"test/parse/precedence.as:4.2-3","text":"x"}},{"tag":"Dereference","span":"test/parse/precedence.as:5.1-3","operand":{"tag":"Identifier","span":"test/parse/precedence.as:5.2-3","text":"x"}},{"tag":"AddressOf","span":"test/parse/precedence.as:6.1-3","mutable":false,"operand":{"tag":"Identifier","span":"test/parse/precedence.as:6.2-3","text":"x"}},{"tag":"Mul","span":"test/parse/precedence.as:9.1-7","lhs":{"tag":"Negate","span":"test/parse/precedence.as:9.1-3","operand":{"tag":"Identifier","span":"test/parse/precedence.as:9.2-3","text":"x"}},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:9.6-7","text":"y"}},{"tag":"Div","span":"test/parse/precedence.as:10.1-7","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:10.1-2","text":"y"},"rhs":{"tag":"Identity","span":"test/parse/precedence.as:10.5-7","operand":{"tag":"Identifier","span":"test/parse/precedence.as:10.6-7","text":"x"}}},{"tag":"Mod","span":"test/parse/precedence.as:11.1-7","lhs":{"tag":"BitNot","span":"test/parse/precedence.as:11.1-3","operand":{"tag":"Identifier","span":"test/parse/precedence.as:11.2-3","text":"y"}},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:11.6-7","text":"x"}},{"tag":"Add","span":"test/parse/precedence.as:14.1-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:14.1-2","text":"y"},"rhs":{"tag":"Mul","span":"test/parse/precedence.as:14.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:14.5-6","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:14.9-10","text":"z"}}},{"tag":"Sub","span":"test/parse/precedence.as:15.1-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:15.1-2","text":"y"},"rhs":{"tag":"Mod","span":"test/parse/precedence.as:15.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:15.5-6","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:15.9-10","text":"z"}}},{"tag":"Div","span":"test/parse/precedence.as:16.2-12","lhs":{"tag":"Sub","span":"test/parse/precedence.as:16.2-7","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:16.2-3","text":"a"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:16.6-7","text":"b"}},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:16.11-12","text":"y"}},{"tag":"BitAnd","span":"test/parse/precedence.as:19.1-6","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:19.1-2","text":"a"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:19.5-6","text":"b"}},{"tag":"BitAnd","span":"test/parse/precedence.as:20.1-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:20.1-2","text":"a"},"rhs":{"tag":"Mul","span":"test/parse/precedence.as:20.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:20.5-6","text":"b"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:20.9-10","text":"x"}}},{"tag":"Mul","span":"test/parse/precedence.as:21.2-12","lhs":{"tag":"BitAnd","span":"test/parse/precedence.as:21.2-7","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:21.2-3","text":"a"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:21.6-7","text":"b"}},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:21.11-12","text":"x"}},{"tag":"BitXor","span":"test/parse/precedence.as:24.1-6","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:24.1-2","text":"a"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:24.5-6","text":"b"}},{"tag":"BitXor","span":"test/parse/precedence.as:25.1-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:25.1-2","text":"b"},"rhs":{"tag":"BitAnd","span":"test/parse/precedence.as:25.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:25.5-6","text":"c"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:25.9-10","text":"d"}}},{"tag":"BitOr","span":"test/parse/precedence.as:28.1-10","lhs":{"tag":"BitXor","span":"test/parse/precedence.as:28.1-6","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:28.1-2","text":"a"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:28.5-6","text":"b"}},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:28.9-10","text":"c"}},{"tag":"BitOr","span":"test/parse/precedence.as:29.1-14","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:29.1-2","text":"b"},"rhs":{"tag":"BitXor","span":"test/parse/precedence.as:29.5-14","lhs":{"tag":"BitAnd","span":"test/parse/precedence.as:29.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:29.5-6","text":"c"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:29.9-10","text":"b"}},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:29.13-14","text":"a"}}},{"tag":"GreaterThan","span":"test/parse/precedence.as:32.1-6","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:32.1-2","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:32.5-6","text":"y"}},{"tag":"GreaterThanOrEqualTo","span":"test/parse/precedence.as:33.1-7","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:33.1-2","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:33.6-7","text":"y"}},{"tag":"GreaterThan","span":"test/parse/precedence.as:36.1-6","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:36.1-2","text":"y"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:36.5-6","text":"x"}},{"tag":"GreaterThanOrEqualTo","span":"test/parse/precedence.as:37.1-7","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:37.1-2","text":"y"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:37.6-7","text":"x"}},{"tag":"EqualTo","span":"test/parse/precedence.as:40.1-7","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:40.1-2","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:40.6-7","text":"y"}},{"tag":"Not","span":"test/parse/precedence.as:43.1-11","operand":{"tag":"EqualTo","span":"test/parse/precedence.as:43.5-11","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:43.5-6","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:43.10-11","text":"y"}}},{"tag":"Not","span":"test/parse/precedence.as:44.1-10","operand":{"tag":"GreaterThan","span":"test/parse/precedence.as:44.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:44.5-6","text":"y"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:44.9-10","text":"x"}}},{"tag":"And","span":"test/parse/precedence.as:47.1-20","lhs":{"tag":"Not","span":"test/parse/precedence.as:47.1-10","operand":{"tag":"LessThan","span":"test/parse/precedence.as:47.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:47.5-6","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:47.9-10","text":"y"}}},"rhs":{"tag":"Boolean","span":"test/parse/precedence.as:47.15-20","value":false}},{"tag":"Or","span":"test/parse/precedence.as:50.1-19","lhs":{"tag":"Not","span":"test/parse/precedence.as:50.1-10","operand":{"tag":"LessThan","span":"test/parse/precedence.as:50.5-10","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:50.5-6","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:50.9-10","text":"y"}}},"rhs":{"tag":"Boolean","span":"test/parse/precedence.as:50.14-19","value":false}},{"tag":"Or","span":"test/parse/precedence.as:51.1-25","lhs":{"tag":"GreaterThan","span":"test/parse/precedence.as:51.1-6","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:51.1-2","text":"y"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:51.5-6","text":"x"}},"rhs":{"tag":"And","span":"test/parse/precedence.as:51.10-25","lhs":{"tag":"LessThan","span":"test/parse/precedence.as:51.10-15","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:51.10-11","text":"x"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:51.14-15","text":"y"}},"rhs":{"tag":"GreaterThan","span":"test/parse/precedence.as:51.20-25","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:51.20-21","text":"y"},"rhs":{"tag":"Integer","span":"test/parse/precedence.as:51.24-25","text":"1"}}}},{"tag":"Assign","span":"test/parse/precedence.as:54.1-11","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:54.1-2","text":"x"},"rhs":{"tag":"Or","span":"test/parse/precedence.as:54.5-11","lhs":{"tag":"Identifier","span":"test/parse/precedence.as:54.5-6","text":"y"},"rhs":{"tag":"Identifier","span":"test/parse/precedence.as:54.10-11","text":"z"}}}]}}
